### PR TITLE
Styling for the Navigation Bar

### PR DESCRIPTION
**Pull request by safchow**
## Trello ticket link

[Developer Task: Implement UI/UX feedback to the navigation bar](https://trello.com/c/5IA61xjQ/119-developer-task-implement-ui-ux-feedback-to-the-navigation-bar )

## Implementation description
* Added new component framework, Chakra UI
* Rebuilt the Navigation Bar

## Steps to test
Routing Tests
1. Run `npm i` to install new deps
2. Verify that all menu options open their corresponding menu lists
3. Verify that each menu list option routes to the correct page
4. Verify there are no visual conflicts with other pages

Style Tests
1. Verify that hovering over menu items changes the background colour to #164684
2. Verify that clicking on a menu item changes the background colour to #214E89
3. Verify that hovering over a menu list item changes the font colour to #023679
4. Optional: ensure styling matches up with the [Figma](https://www.figma.com/file/9HWk5TlYt0JyuBmSKRskeC/Safwaan-Schoel-Preojnfnct?node-id=1-4&t=ni9RP11P2Jp9ZeCU-0 ) mock

Visual Tests
1. Verify all margins are correct, including margins between menu items and icons (such as chevrons or add icons)
2. Verify that all pages that display Alerts function correctly

## What should reviewers focus on?
* Are there colour schemes we should adopt? Fonts and font sizes?

### CODE REVIEW

File: src/components/events/Events.js
Comment (gabbytang) [inline src/components/events/Events.js:1]: When I try to join an event, the page doesn't give a toast message or reload. It just goes completely blank. It doesn't matter if I'm already a participant or not.
If I manually reload the webpage, it appears Im still able to join the list of participants.

Comment (gabbytang) [inline src/components/events/Events.js:1]: I know you didn't make any changes to the file itself, but the screen also goes blank after I create an event.

File: src/components/forums/MyForums.js
Comment (gabbytang) [inline src/components/forums/MyForums.js:1]: Screen also goes blank when I try to archive a forum. There;s something screwing with some window reloads, but only in some cases. When I added a comment to a forum the page reloaded just fine, with no errors.

File: src/components/common/NavigationBar.js [hunk 1]
```diff
@@ -1,340 +1,214 @@
...
-                                        mr: 2,
-                                        display: { xs: 'none', md: 'flex' },
-                                        fontFamily: 'monospace',
-                                        fontWeight: 700,
-                                        letterSpacing: '.3rem',
-                                        color: 'inherit',
-                                        textDecoration: 'none'
-                                    }}
-                                >
-                                    HonkedIn
```
Comment (gabbytang) [inline src/components/common/NavigationBar.js:178]: As a UI aspect thats typical in a lot of cases, what do you think about linking the home page to the "HonkedIn" logo/word also. As in keep the home button, but also link home in the logo. I think LinkedIn does that

File: src/components/resources/Resources.js
Comment (gabbytang) [inline src/components/resources/Resources.js:1]: Getting same reload error for creation of a resource

**Review by gabbytang — CHANGES_REQUESTED** [review]
The styling is a major improvement from what we had before, its so aesthetically pleasing to look at the menu options. There was one major issues I was getting which tbh doesn't match the changes you've made. You should take a look if you are getting the same issues, because often times after completing a function, the screen doesn't reload, and is instead a blank screen. I think its an issue with toast messages.

**Comment by safchow** [comment]
> When I try to join an event, the page doesn't give a toast message or reload. It just goes completely blank. It doesn't matter if I'm already a participant or not. If I manually reload the webpage, it appears Im still able to join the list of participants.

> Screen also goes blank when I try to archive a forum. There;s something screwing with some window reloads, but only in some cases. When I added a comment to a forum the page reloaded just fine, with no errors.

> Getting same reload error for creation of a resource

These all seem to be caused by the Alert component, the backend and APIs are still processing requests. There may be a conflict with MUI Alert and the Chakra UI framework. Looking into a fix, might possibly need a refactor of the Alert component

**Comment by safchow** [comment]
> > When I try to join an event, the page doesn't give a toast message or reload. It just goes completely blank. It doesn't matter if I'm already a participant or not. If I manually reload the webpage, it appears Im still able to join the list of participants.
> 
> > Screen also goes blank when I try to archive a forum. There;s something screwing with some window reloads, but only in some cases. When I added a comment to a forum the page reloaded just fine, with no errors.
> 
> > Getting same reload error for creation of a resource
> 
> These all seem to be caused by the Alert component, the backend and APIs are still processing requests. There may be a conflict with MUI Alert and the Chakra UI framework. Looking into a fix, might possibly need a refactor of the Alert component

@gabbytang Good catch, it was a component framework conflict, fix has been put in. Please verify that all the pages that display Alerts function correctly.

**Review by Eshah7 — COMMENTED** [review]
Not quite sure how to explain this...

However, when I hover over the "HonkedIn" text, the cursor changes into the "I", editing symbol rather than indicating to the user that they are able to click on the HonkedIn text.

**Comment by Eshah7** [comment]
Screenshot 2023-03-18 at 6 22 28 PM

Is the toast message supposed to have any space from the bottom, similar to what we had before?

**Comment by Eshah7** [comment]
Not sure if this is addressed in the same PR but, I was thinking if we could replace the gear symbol with a silhouette of a person because it is a little confusing as to where I can view my "own profile"!

**Comment by Eshah7** [comment]
Screenshot 2023-03-18 at 6 24 57 PM

Also, I think due to the font change, the spacing for my profile page has been disrupted a little...

**Comment by Eshah7** [comment]
Overall, everything else LGTM, it's really pretty!!! Thank you :)

**Comment by gabbytang** [comment]
> Not sure if this is addressed in the same PR but, I was thinking if we could replace the gear symbol with a silhouette of a person because it is a little confusing as to where I can view my "own profile"!

I actually don't mind the gear, I feel like a lot of sites and apps have edit profile under settings.

**Comment by gabbytang** [comment]
> Not quite sure how to explain this...
> 
> However, when I hover over the "HonkedIn" text, the cursor changes into the "I", editing symbol rather than indicating to the user that they are able to click on the HonkedIn text.

Yeah, when you hover over the HonkedIn icon the cursor is not a finger pointing which typically is what indicates a link.

**Comment by gabbytang** [comment]
Once you implement Esha's changes, i think everything looks great. Nice work on cleaning up our nav bar, it really makes our site look more professional

**Review by gabbytang — APPROVED** [review]

**Comment by safchow** [comment]
> Screenshot 2023-03-18 at 6 22 28 PM
> 
> Is the toast message supposed to have any space from the bottom, similar to what we had before?

its currently fixed to the bottom right, I can add margins from the corners if that's preferred

**Comment by safchow** [comment]
> Not quite sure how to explain this...
> 
> However, when I hover over the "HonkedIn" text, the cursor changes into the "I", editing symbol rather than indicating to the user that they are able to click on the HonkedIn text.

I changed the component to a Link, let me know if this fixes the issue

**Comment by safchow** [comment]
> Screenshot 2023-03-18 at 6 24 57 PM
> 
> Also, I think due to the font change, the spacing for my profile page has been disrupted a little...

This isn't related to the font change, as only the navbar's font has been updated. It's being caused by this:
```
            <Box sx={{ position: 'absolute', top: 240, left: "35%" }} >
```

Updating how many pixels this component should render from the top should resolve this issue, but this page needs a UI overhaul, so it can be addressed then

**Comment by Eshah7** [comment]
> > Screenshot 2023-03-18 at 6 22 28 PM
> > Is the toast message supposed to have any space from the bottom, similar to what we had before?
> 
> its currently fixed to the bottom right, I can add margins from the corners if that's preferred

I think margins would be nice, however not really a high priority! Thank you :)

**Comment by Eshah7** [comment]
> > Not quite sure how to explain this...
> > However, when I hover over the "HonkedIn" text, the cursor changes into the "I", editing symbol rather than indicating to the user that they are able to click on the HonkedIn text.
> 
> I changed the component to a Link, let me know if this fixes the issue

Fixed! Thanks :)

**Review by Eshah7 — APPROVED** [review]
Lgtm!

**Pull request merged**